tools/nixpkgs-hammer

```diff
@@ -1,14 +1,46 @@
 #!/usr/bin/env python3
 
+import os
 from pathlib import Path
 import argparse
 import json
 import subprocess
 import sys
 import textwrap
+from typing import List
+from collections import defaultdict
 
 
-NIX_INSTANTIATE_PATH = 'nix-instantiate'
+def ast_checks(nix_file: str, attrs: List[str], excluded_rules: List[str]):
+    env = dict(os.environ)
+    env["EDITOR"] = "echo"
+
+    file_to_attr = defaultdict(list)
+    for attr in attrs:
+        proc = subprocess.run(
+            ["nix", "edit", "-f", nix_file, attr],
+            env=env, stdout=subprocess.PIPE, text=True, check=True)
+        file_to_attr[proc.stdout.strip()].append(attr)
+
+    # Each rule is the name of a binary on our PATH
+    rules = {"missing-patch-comment"} - set(excluded_rules)
```

Comment:
Hmm, could we have this list auto-generated by something (e.g. the in `flake.nix` and substituted in).